src/Microsoft.OData.Core/Json/JsonReader.cs

```diff
@@ -363,6 +434,56 @@ public virtual bool Read()
             return true;
         }
 
+        /// <summary>
+        /// Creates a stream for reading a base64 binary value.
+        /// </summary>
+        /// <returns>A stream for reading a base64 URL encoded binary value.</returns>
+        public Stream CreateReadStream()
+        {
+            if (!this.canStream)
+            {
+                throw new Exception("CreateTextReader cannot be called in the current state");
+            }
+
+            this.canStream = false;
+            if ((this.streamOpeningQuoteCharacter = characterBuffer[this.tokenStartIndex]) == 'n')
+            {
+                this.ParseNullPrimitiveValue();
+                this.scopes.Peek().ValueCount++;
+                this.Read();
+                return new ODataBinaryStreamReader((a, b, c) => { return 0; });
+            }
+
+            this.tokenStartIndex++;

```

Comment:
>this.tokenStartIndex++; [](start = 12, length = 23)

what about if "startIndex" point to the last valid character in the buffer?

Reply:
On line 961, before doing anything else, ReadChars checks to see if tokenStartIndex < storedCharacterCount. If not, it calls ReadInput().

---
In reply to: [258647724](https://github.com/OData/odata.net/pull/1204#discussion_r258647724) [](ancestors = 258647724)